view/components/head/script.dust

```diff
@@ -10,15 +10,66 @@
 
     window.appInsights=appInsights,appInsights.queue&&0===appInsights.queue.length;
 
-    appInsights.queue.push(function() {
-      requestIDNode = document.head.querySelector("meta[name=applicationinsights-request-id]");
-      if(requestIDNode) {
-        appInsights.context.operation.id = requestIDNode.content;
+    {! If the applicationinsights-request-id meta tag is present, we add a function to the Appinsights queue which runs before our pageView and allows us to track events through the whole stack. !}
+    var requestIDNode = document.head.querySelector("meta[name=applicationinsights-request-id]");
+    if(requestIDNode) {
+      appInsights.queue.push(function() {
+        var parentID, firstChar, lastChar, generatedID, operationID;
+        parentID = requestIDNode.content;
+
+        firstChar = parentID.charAt(0);
+        if(firstChar != '|') {
+          return;
+        }
+
+        lastChar = parentID.charAt(parentID.length - 1);
+        if(lastChar != '.' && lastChar != '_') {
+          parentID += '.';
+        }
+
+        generatedID = ukparliamentAppinsightsRandomIdGenerator();
+        parentID += generatedID + '.';
+
+        operationID = ukparliamentAppinsightsGetOperationId(parentID);
+
+        appInsights.context.operation.id = operationID;
+        appInsights.context.operation.parentID = parentID;
```

Comment:
`parentID` property isn't available, seems to be because of this:
```
if(firstChar != '|') {
  return;
}
```
Not sure if this is correct?

Reply:
The parent ID should be in the format: `|abcd1234.` or `|abcd1234_` - if the request ID doesn't start with a | then we skip execution.

Microsoft's sample code changes from implementation to implementation, some checking for `|` and some ignoring - if you'd like I can skip?

Reply:
looks good 👍